### PR TITLE
3d marker pose

### PR DESCRIPTION
**Pull request by marc-tonsen**
Added a function to compute the 3D pose of a calibration marker given it's 2D description

### CODE REVIEW

File: pupil_src/shared_modules/circle_detector.py [hunk 1]
```diff
@@ -464,6 +465,46 @@ def get_nested_clusters(contours, hierarchy, min_nested_count):
...
+    return pts_rot
+
+def marker_3d_pose(marker, cam_model, marker_diameter=7.6):
+
+    target_circle = [[0, 0], [marker_diameter, marker_diameter], 0]
+    target_pts = getEllipsePts(target_circle)
+    target_pts3D = np.zeros((target_pts.shape[0], target_pts.shape[1] + 1))
+    target_pts3D[:, :-1] = target_pts
+    target_pts3D.shape = -1, 1, 3
+    target_pts3D = target_pts3D.astype('float32')
```
Comment (papr) [inline pupil_src/shared_modules/circle_detector.py]: The type could be defined on creation (np.zeros). This way you save one copy.

File: pupil_src/shared_modules/circle_detector.py [hunk 2]
```diff
@@ -464,6 +465,46 @@ def get_nested_clusters(contours, hierarchy, min_nested_count):
             clusters[cluster[1]] = cluster
     return clusters.values()
 
+def getEllipsePts(e, num_pts=10):
```
Comment (mkassner) [inline pupil_src/shared_modules/circle_detector.py]: can we not use this here? https://docs.opencv.org/3.0-beta/modules/imgproc/doc/drawing_functions.html#ellipse2poly

File: pupil_src/shared_modules/circle_detector.py (hunk 1 above)
Comment (marc-tonsen) [reply pupil_src/shared_modules/circle_detector.py]: ok, I changed that!

File: pupil_src/shared_modules/circle_detector.py (hunk 2 above)
Comment (mkassner) [reply pupil_src/shared_modules/circle_detector.py]: We now know this fn sucks.

**Pull request merged**